packages/eslint-plugin/docs/rules/no-misused-spread.mdx

```diff
@@ -95,6 +181,41 @@ If you want to allow this behavior, you can configure the rule with `"allowClass
 }
 ```
 
+### `allowClassDeclarations`
+
+By default, this rule disallows using the spread operator on class declarations:
```

Comment:
I don't like the name of this option, as the class could also be passed as a reference or be generated by a class factory.
I asked ChatGPT for a better name and it suggested "class reference" as an alternative, WDYT?

Reply:
I personally don't have any strong opinion about either of the names

Maybe let's wait for one of the maintainers to decide?

Reply:
I would suggest `allowClasses`. Any concerns?

Reply:
@Josh-Cena It's better than declaration, but `allowClasses ` can be understood (or confused) as class instances which is another option.

Reply:
Right, but "class instances" and "classes" are obviously different things and the fact that we already have an "instances" option further mitigates that.

Reply:
Well, yes, assuming you read about both in proximity, seeing an eslint-disable comment or config option can lose the context

Reply:
Also classes are just sugared classes so I'm not sure if we want to treat them any more specially than functions or anything with a `new` signature...